### PR TITLE
Introduce purs-tidy formatter

### PR DESCRIPTION
**Pull request by thomashoneyman**
**Description of the change**
Introduces the [`purs-tidy`](https://github.com/natefaubion/purescript-tidy) formatter. This formatter is a lightly opinionated tool we use to maintain a consistent style in the contrib libraries.

We ordinarily restrict to tools provided by the `core` or `contrib` projects. This tool is not, but it was developed and is maintained by core team members, and because it's a formatter it can't block maintenance or release of this library even in the event something goes catastrophically wrong with it.

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a link to this PR and your username
- [x] Linked any existing issues or proposals that this pull request should close
- [x] Updated or added relevant documentation in the README and/or documentation directory
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

File: src/Concurrent/BoundedQueue.purs [hunk 1]
```diff
@@ -78,11 +78,11 @@ tryRead (BoundedQueue q) = do
...
 -- | block.
-tryWrite ∷ ∀ a. BoundedQueue a → a → Aff Boolean
+tryWrite :: forall a. BoundedQueue a -> a -> Aff Boolean
 tryWrite (BoundedQueue q) a = do
-  w ← AVar.take q.writePos
-  AVar.tryPut a (unsafePartial unsafeIndex q.contents w) >>= if _
-    then do
+  w <- AVar.take q.writePos
+  AVar.tryPut a (unsafePartial unsafeIndex q.contents w) >>=
+    if _ then do
```
Comment (JordanMartinez) [inline src/Concurrent/BoundedQueue.purs:85]: Thoughts on this style?

File: test/BoundedQueue.purs [hunk 2]
```diff
@@ -15,156 +15,160 @@ import Effect.Class (liftEffect)
...
       r1 `shouldEqual` (Just 1)
       assert "Should've been Nothing" (isNothing r2)
-    test ("(Sync) tryRead does not block when there are consumers blocked on "
-      <> "the queue") $ liftAff do
-      q ← liftEffect (BQS.new 1)
-      _ ← forkAff (BQ.read q)
-      r ← race (delayMs 20) (liftEffect (BQS.tryRead q))
+    test
+      ( "(Sync) tryRead does not block when there are consumers blocked on "
+          <> "the queue"
```
Comment (JordanMartinez) [inline test/BoundedQueue.purs]: Should this `<>` be removed?

File: test/BoundedQueue.purs [hunk 3]
```diff
@@ -15,156 +15,160 @@ import Effect.Class (liftEffect)
...
-      r ← BQ.read q
+      r <- BQ.read q
       assert "tryWrite should've succeeded" rw
       r `shouldEqual` 1
-    test ("(Sync) tryWrite does not block when there are writers blocked on " <>
-      "the queue") $ liftAff do
-      q ← liftEffect (BQS.new 1)
+    test
+      ( "(Sync) tryWrite does not block when there are writers blocked on " <>
+          "the queue"
```
Comment (JordanMartinez) [inline test/BoundedQueue.purs]: Same here.

**Review by JordanMartinez — APPROVED** [review]
LGTM besides the comments above.

File: src/Concurrent/BoundedQueue.purs (hunk 1 above)
Comment (thomashoneyman) [reply src/Concurrent/BoundedQueue.purs:85]: I'm personally fine with it!

**Pull request merged**